### PR TITLE
Fix fp16 conversion from and to fp32 for Nan/Inf kind of values.

### PR DESCRIPTION
**Pull request by ThomasRaoux**

### CODE REVIEW

File: runtime/src/iree/base/internal/math.h [hunk 1]
```diff
@@ -277,7 +277,13 @@ static inline float iree_math_f16_to_f32(const uint16_t f16_value) {
   const uint32_t sign = ((uint32_t)((f16_value & 0x8000u) >> 15)) << 31;
   uint32_t exp = ((f16_value & 0x7C00u) >> 10);
   uint32_t mantissa = 0;
-  if (exp > 0) {
+  if (exp == 0x1F) {
```
Comment (antiagainst) [inline runtime/src/iree/base/internal/math.h:280]: Should we also update the `NOTE` in the above comment?

Comment (antiagainst) [inline runtime/src/iree/base/internal/math.h:280]: Nit: `0x1Fu` to follow conventions used for other constants?

Similarly for the following.

File: runtime/src/iree/base/internal/math.h [hunk 2]
```diff
@@ -277,7 +277,13 @@ static inline float iree_math_f16_to_f32(const uint16_t f16_value) {
...
+    // Nan or Inf case.
```
Comment (antiagainst) [inline runtime/src/iree/base/internal/math.h:281]: Nit: s/Nan/NaN/

Similar for the following.

File: runtime/src/iree/base/internal/math.h [hunk 3]
```diff
@@ -295,10 +301,18 @@ static inline uint16_t iree_math_f32_to_f16(const float f32_value) {
   uint32_t u32_value;
   memcpy(&u32_value, &f32_value, sizeof(u32_value));
   const uint32_t sign = ((u32_value & 0x80000000u) >> 31) << 15;
-  const uint32_t mantissa = (u32_value & 0x007FFFFFu) >> (23 - 10);
+  uint32_t mantissa = (u32_value & 0x007FFFFFu) >> (23 - 10);
   int32_t exp = ((u32_value & 0x7F800000u) >> 23) - 127 + 15;
   if (exp > 31) {
     exp = 31 << 10;
+    // zero out the matissa for infinity.
```
Comment (antiagainst) [inline runtime/src/iree/base/internal/math.h:308]: typo; should be "mantissa".

**Review by antiagainst — APPROVED** [review]
Cool! I LGTM; just a few nits.

File: runtime/src/iree/base/internal/math.h (hunk 1 above)
Comment (ThomasRaoux) [reply runtime/src/iree/base/internal/math.h]: I made some small change to the NOTE but I'm not convinced we handle all the NaN cases.

**Pull request merged**